src/stan/lang/grammars/semantic_actions_def.cpp

```diff
@@ -547,13 +541,16 @@ namespace stan {
       // build up representations
       expr_type result_type(decl.return_type_.base_type_,
                             decl.return_type_.num_dims_);
-      std::vector<expr_type> arg_types;
+      std::vector<function_arg_type> arg_types;
       for (size_t i = 0; i < decl.arg_decls_.size(); ++i)
-        arg_types.push_back(expr_type(decl.arg_decls_[i].arg_type_.base_type_,
-                                      decl.arg_decls_[i].arg_type_.num_dims_));
+        arg_types.push_back(
+            function_arg_type(expr_type(decl.arg_decls_[i].arg_type_.base_type_,
```

Comment:
When this gets updated with the new `expr_type` branch, this argument won't be primitive, so you'll need to set a local variable in order to pass it in by reference.  Otherwise, g++ will fail, because it clears memory more agressively than clang++.

Reply:
noted, thanks.